flocker/node/script.py

```diff
@@ -50,14 +52,24 @@ class ZFSAgentOptions(Options):
          "The port on the control service to connect to.", int],
     ]
 
-    def parseArgs(self, hostname, host):
-        # Passing in the 'hostname' (really node identity) via command
-        # line is a hack.  See
-        # https://clusterhq.atlassian.net/browse/FLOC-1381 for solution.
-        self["hostname"] = unicode(hostname, "ascii")
+    def parseArgs(self, host):
         self["destination-host"] = unicode(host, "ascii")
 
 
+def _get_external_ip(host, port):
+    sock = socket()
+    sock.connect_ex((host, port))
+    return sock.getaddr()[1]  # XXX approximately
+
+
+def _get_node_uuid(ip):
+    # Soon we'll extract this from TLS certificate for node.  Until then
+    # we'll just do a temporary hack of some sort.  In particular, by
+    # using a deterministic IP->UUID mapping we can separate out porting
+    # the REST API to new UUID model into a separate issue.
+    pass #return UUID(md5(ip))
```

Comment:
I thought it might be handy if we could do `UUID(node=ip, everything_else=0)` so the mapping could be easily reversible but it seems like the `node` component doesn't map easily to an IP address and there's no easy way to do construct a `UUID` instance from components with the `uuid` module anyway.